src/summary.rs

```diff
@@ -40,39 +40,114 @@ impl std::fmt::Display for Rewards {
     }
 }
 
-/// Struct for holding the information of what to be displayed with the `info`
-/// command
-#[derive(Deserialize, Serialize, Debug)]
+/// struct for flexibly updating the fields of the summary
+#[derive(Default, Debug)]
+pub(crate) struct SummaryUpdateFields {
+    pub(crate) is_plotting_finished: bool,
+    pub(crate) is_new_block_farmed: bool,
+    pub(crate) is_new_vote: bool,
+    pub(crate) maybe_new_reward: Option<Rewards>,
+    pub(crate) maybe_last_block: Option<u64>,
```

Comment:
```suggestion
    pub(crate) is_plotting_finished: bool,
    pub(crate) new_block_farmed: u32,
    pub(crate) new_vote: bool,
    pub(crate) new_rewards: Rewards,
    pub(crate) maybe_last_block: Option<u64>,
```

Reply:
we already have this struct as `Summary`
this struct was intermediate for updating the summary.
After our discussion, it probably won't be used